pkg/sources/adapter/googlecloudpubsubsource/process.go

```diff
@@ -76,93 +74,23 @@ func makePubSubEvent(msg *pubsub.Message, srcAttr string) (*cloudevents.Event, e
 // JSON serialization inside some CloudEvent data.
 func toCloudEventData(msg *pubsub.Message) interface{} {
 	var data interface{}
-	data = msg.Data
+	data = msg
```

Comment:
The main change in the PR is actually **here**, in this very simple change.
Instead of propagating the message's data and filtering every other field, propagate the Pub/Sub message untouched.